integration-test/mqtt/mqtt_system_test.c

```diff
@@ -819,10 +821,14 @@ void tearDown()
     }
 
     /* Terminate MQTT connection. */
-    TEST_ASSERT_EQUAL( MQTTSuccess, MQTT_Disconnect( &context ) );
+    status = MQTT_Disconnect( &context );
 
     /* Terminate TLS session and TCP connection. */
     ( void ) Openssl_Disconnect( &networkContext );
+
+    /* Make any assertions at the end that may prevent #Openssl_Disconnect from
```

Comment:
Why are we throwing away the return value from `Openssl_Disconnect()`?  Shouldn't you also validate that in the same way that you're validating the result from `MQTT_Disconnect()`?  If you did this then you'd know that it `Openssl_Disconnect()` has been called and you could delete this comment.